mmdet/evaluation/metrics/coco_metric.py

```diff
@@ -341,217 +112,150 @@ def process(self, data_batch: dict, data_samples: Sequence[dict]) -> None:
-                    '`ann_file` is not provided'
-                gt['anns'] = data_sample['instances']
-            # add converted result to the results list
-            self.results.append((gt, result))
-
-    def compute_metrics(self, results: list) -> Dict[str, float]:
-        """Compute the metrics from processed results.
-
-        Args:
-            results (list): The processed results of each batch.
-
-        Returns:
-            Dict[str, float]: The computed metrics. The keys are the names of
-            the metrics, and the values are corresponding results.
+                ann = self.add_gt(data_sample)
+            else:
+                ann = dict()
+            groundtruths.append(ann)
+
+        self.add(predictions, groundtruths)
+
+    @staticmethod
+    def add_gt(data_sample):
+        ann = dict()
+        ann['width'] = data_sample['ori_shape'][1]
+        ann['height'] = data_sample['ori_shape'][0]
+        ann['img_id'] = data_sample['img_id']
+
+        gt_instances = data_sample['gt_instances']
+        ignored_instances = data_sample['ignored_instances']
+
+        ann['bboxes'] = np.concatenate(
+            (gt_instances['bboxes'].cpu().numpy(),
+             ignored_instances['bboxes'].cpu().numpy()),
+            axis=0)
+        ann['labels'] = np.concatenate(
+            (gt_instances['labels'].cpu().numpy(),
+             ignored_instances['labels'].cpu().numpy()),
+            axis=0)
+        ann['ignore_flags'] = np.concatenate(
+            (np.zeros(len(gt_instances['labels'])),
+             np.ones(len(ignored_instances['labels']))),
+            axis=0)
+        assert len(ann['bboxes']) == len(ann['labels'])
+        if 'masks' in gt_instances:
+            assert isinstance(gt_instances['masks'],
+                              (PolygonMasks, BitmapMasks)) and \
+                   isinstance(ignored_instances['masks'],
+                              (PolygonMasks, BitmapMasks))
+            ann['masks']: list = []
+            ann['masks'].extend(
+                encode_mask_results(gt_instances['masks'].to_ndarray()))
+            ann['masks'].extend(
+                encode_mask_results(ignored_instances['masks'].to_ndarray()))
+            assert len(ann['bboxes']) == len(ann['masks'])
+        return ann
+
+    def evaluate(self, *args, **kwargs) -> dict:
+        """Returns metric results and print pretty table of metrics per class.
+
+        This method would be invoked by ``mmengine.Evaluator``.
         """
-        logger: MMLogger = MMLogger.get_current_instance()
-
-        # split gt and prediction list
-        gts, preds = zip(*results)
-
-        tmp_dir = None
-        if self.outfile_prefix is None:
-            tmp_dir = tempfile.TemporaryDirectory()
-            outfile_prefix = osp.join(tmp_dir.name, 'results')
-        else:
-            outfile_prefix = self.outfile_prefix
+        metric_results = self.compute(*args, **kwargs)
```

Comment:
Does it support saving coco summarize information?

https://github.com/open-mmlab/mmdetection/blob/c14dd6c42efb63f662a63fe403198bac82f47aa6/mmdet/datasets/coco.py#L514-L519

Reply:
Seems not supported, but we can save them in list and print it as a table

Reply:
![image](https://user-images.githubusercontent.com/48282753/221537172-dcc45107-f67c-4bc5-85fc-d28f9497b4c0.png)
Supporting Now